tests/unit/extractor/dashboard/tableau/test_tableau_dashboard_last_modified_extractor.py

```diff
@@ -0,0 +1,67 @@
+# Copyright Contributors to the Amundsen project.
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+import unittest
+
+from mock import patch
+from pyhocon import ConfigFactory  # noqa: F401
+
+from databuilder import Scoped
+from databuilder.extractor.dashboard.tableau.tableau_dashboard_last_modified_extractor import TableauDashboardLastModifiedExtractor
+from databuilder.extractor.dashboard.tableau.tableau_dashboard_utils import TableauDashboardAuth, TableauDashboardUtils, TableauGraphQLApiExtractor
+from databuilder.models.dashboard.dashboard_last_modified import DashboardLastModifiedTimestamp
+
+
+logging.basicConfig(level=logging.INFO)
+
+
+def mock_query(*args, **kwargs):
+    return {
+        'workbooks': [
+          {
+            "id": "fake-workbook-id",
+            "name": "Test Workbook",
+            "projectName": "Test Project",
+            "updatedAt": "2020-08-04T20:16:05Z"
+          }
+        ]
+      }
+
+def mock_token(*args, **kwargs):
+    return '123-abc'
+
+class TestTableauDashboardLastModified(unittest.TestCase):
+
+    @patch.object(TableauDashboardAuth, '_authenticate', mock_token)
+    @patch.object(TableauGraphQLApiExtractor, 'execute_query', mock_query)
+    def test_dashboard_last_modified_extractor(self):
+
+        config = ConfigFactory.from_dict({
+            'extractor.tableau_dashboard_last_modified.tableau_host': 'tableau_host',
+            'extractor.tableau_dashboard_last_modified.api_version': 'tableau_api_version',
+            'extractor.tableau_dashboard_last_modified.site_name': 'tableau_site_name',
+            'extractor.tableau_dashboard_last_modified.tableau_personal_access_token_name':
+                'tableau_personal_access_token_name',
+            'extractor.tableau_dashboard_last_modified.tableau_personal_access_token_secret':
+                'tableau_personal_access_token_secret',
+            'extractor.tableau_dashboard_last_modified.excluded_projects': [],
+            'extractor.tableau_dashboard_last_modified.cluster': 'tableau_dashboard_cluster',
+            'extractor.tableau_dashboard_last_modified.database': 'tableau_dashboard_database',
+            'extractor.tableau_dashboard_last_modified.transformer.timestamp_str_to_epoch.timestamp_format':
+                '%Y-%m-%dT%H:%M:%SZ',
+
+        })
+
+        extractor = TableauDashboardLastModifiedExtractor()
+        extractor.init(Scoped.get_scoped_conf(conf=config, scope=extractor.get_scope()))
+        record = extractor.extract()
+
+        self.assertEqual(record._dashboard_id, 'Test Workbook')
+        self.assertEqual(record._dashboard_group_id, 'Test Project')
+        self.assertEqual(record._product, 'tableau')
+        self.assertEqual(record._cluster, 'tableau_dashboard_cluster')
+        self.assertEqual(record._last_modified_timestamp, 1596572165)
```

Comment:
This works, but we shouldn't be accessing private attributes of the record. Instead, the assertion should look something like this where we're verifying against a public interface (repr).

```python
        expected = DashboardLastModifiedTimestamp(
            dashboard_group_id='Test Project',
            dashboard_id='Test Workbook',
            ...)
        self.assertEqual(record.__repr__(), expected.__repr__())
```

Reply:
Good callout. I'm not super crazy about `repr()` as the sole interface but I agree it's better than testing the private attributes of a class. +1

Reply:
However, for some of the other models like `DashboardMetadata`, none of the data attributes are prefixed with a `_`, leading me to believe that those aren't intended to be private. So it seems weird to me that the data attributes for `DashboardLastModifiedTimestamp` are private when those aren't - maybe there's a consistency issue upstream?

Reply:
Oh, didn't notice that. Yeah, probably just a mistake.